include/particles/particle.tcc

```diff
@@ -115,8 +115,9 @@ bool mpm::Particle<Tdim>::initialise_particle(
       auto mat_state_vars = material_->initialise_state_variables();
       if (mat_state_vars.size() == particle.nstate_vars) {
         unsigned i = 0;
-        for (const auto& mat_state_var : mat_state_vars) {
-          this->state_variables_[mat_state_var.first] = particle.svars[i];
+        auto state_variables = material_->state_variables();
+        for (const auto& state_var : state_variables) {
+          this->state_variables_.at(state_var) = particle.svars[i];
```

Comment:
This works! I have tried with my case and they are giving the right order of state variables